baseapp/options.go

```diff
@@ -232,14 +236,24 @@ func (app *BaseApp) SetInterfaceRegistry(registry types.InterfaceRegistry) {
 }
 
 // SetStreamingService is used to set a streaming service into the BaseApp hooks and load the listeners into the multistore
-func (app *BaseApp) SetStreamingService(s StreamingService) {
+func (app *BaseApp) SetStreamingService(
+	appOpts serverTypes.AppOptions,
+	appCodec storetypes.Codec,
+	keys map[string]*storetypes.KVStoreKey) {
+	streamers, _, err := streaming.LoadStreamingServices(appOpts, appCodec, app.logger, keys)
+	if err != nil {
+		app.logger.Error("failed to load state streaming", "err", err)
+		os.Exit(1)
```

Comment:
Personally I don't really like that a setter can exit the app. Maybe it should return an error?
Given that it does more than just setting it would rename the function as well.

Reply:
agree, I can make it return an error